libpromises/cmdb.c

```diff
@@ -503,18 +503,21 @@ bool LoadCMDBData(EvalContext *ctx)
 
     bool success = true;
     JsonElement *vars = JsonObjectGet(data, "vars");
-    if ((vars != NULL) && !ReadCMDBVars(ctx, vars))
+    if ((vars != NULL) && (JsonGetType(vars) != JSON_TYPE_NULL) &&
+        !ReadCMDBVars(ctx, vars))
     {
         success = false;
```

Comment:
What does `success` do here? At first glance, it seems to me that `null` is expected and should not cause `success = false`.

Reply:
That's how it works, though, isn't it? `success` is only set to `false` if there is valid non-null data and processing fails.

Reply:
Ah, yes you are correct. My thinking was inverted.